data-prepper-core/src/test/java/org/opensearch/dataprepper/peerforwarder/RemotePeerForwarderTest.java

```diff
@@ -214,6 +214,36 @@ void test_receiveRecords_should_return_record_from_buffer() throws Exception {
         verify(recordsReceivedFromPeersCounter).increment(3.0);
     }
 
+    @Test
+    void test_receiveRecords_with_no_identification_keys() throws Exception {
+        AggregatedHttpResponse aggregatedHttpResponse = mock(AggregatedHttpResponse.class);
+        when(aggregatedHttpResponse.status()).thenReturn(HttpStatus.OK);
+        when(peerForwarderClient.serializeRecordsAndSendHttpRequest(anyCollection(), anyString(), anyString(), anyString())).thenReturn(aggregatedHttpResponse);
+
+        final List<String> testIps = List.of("8.8.8.8", "127.0.0.1");
+        lenient().when(hashRing.getServerIp(List.of("value1", "value1"))).thenReturn(Optional.of(testIps.get(0)));
+        lenient().when(hashRing.getServerIp(List.of("value2", "value2"))).thenReturn(Optional.of(testIps.get(1)));
+
+        RemotePeerForwarder peerForwarder = createObjectUnderTest();
+        final Collection<Record<Event>> testRecords = generateBatchRecords(2);
+        // Add an event that doesn't have identification keys in it
```

Comment:
Can you provide an example of identification keys ?

Reply:
Need not be customer error. We have a grok processor that extracts fields and if an input that doesn't match the grok comes, it won't have the fields needed for aggregate processor. This kind of inputs should be ignored instead of causing crash.